src/taipy/core/_entity/_submittable.py

```diff
@@ -37,10 +37,59 @@ def submit(
     ):
         raise NotImplementedError
 
-    def _get_inputs(self) -> Set[DataNode]:
-        dag = self._build_dag()
+    def __get_all_data_nodes_in_dag(self, dag: Optional[nx.DiGraph] = None) -> Set[DataNode]:
+        dag = self._build_dag() if not dag else dag
+        return {node for node in dag.nodes if isinstance(node, DataNode)}
+
+    def get_inputs(self, dag: Optional[nx.DiGraph] = None) -> Set[DataNode]:
```

Comment:
Since `get_inputs` is exposed to the end user, I don't think we should accept an `nx.DiGraph` as a parameter.

Maybe you can have something like:
``` python
def get_inputs(self) -> Set[DataNode]:
    dag = self._build_dag()
    return self.__get_inputs(dag)

 def __get_inputs(self, dag: Optional[nx.DiGraph]) -> Set[DataNode]:
    return {node for node, degree in dict(dag.in_degree).items() if degree == 0 and isinstance(node, DataNode)}
```